src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocument.cs

```diff
@@ -57,7 +57,7 @@ public override LSPDocumentSnapshot CurrentSnapshot
             }
         }
 
-        public override LSPDocumentSnapshot UpdateVirtualDocument<TVirtualDocument>(IReadOnlyList<ITextChange> changes, long hostDocumentVersion)
+        public override LSPDocumentSnapshot UpdateVirtualDocument<TVirtualDocument>(IReadOnlyList<ITextChange> changes, int? hostDocumentVersion)
```

Comment:
hostDocumentVersion here should never be `null`. We can't update a virtual document without a corresponding host document version.

All that said, this is technically a breaking change for WTE to absorb. You'll need to duplicate this UpdateVirtualDocument method with an `int` accepting one and obsolete this `long` accepting one. That way WTE can react in parallel instead of blocking your insertion. You'll need to do this for any API breaking changes in the ContainedLanguage assembly.

FYI @ToddGrun / @alexgav / @jimmylewis

Reply:
Wait actually, you've been running this locally. How does this not break WTE already locally? @ToddGrun / @alexgav / @jimmylewis do you all ship your own version of ContainedLanguage?

Reply:
ooo it's because we don't use HTML-delegation yet I bet